R/deploymentTarget.R

```diff
@@ -59,15 +59,31 @@ deploymentTarget <- function(recordPath = ".",
       appDeployments$server
     )
   } else {
-    cli::cli_abort(
-      c(
-        "This app has been previously deployed in multiple places.",
-        "Please use {.arg appName}, {.arg server} or {.arg account} to disambiguate.",
-        i = "Known application names: {.str {unique(appDeployments$name)}}.",
-        i = "Known servers: {.str {unique(appDeployments$server)}}.",
-        i = "Known account names: {.str {unique(appDeployments$account)}}."
-      ),
-      call = error_call
+    choices <- paste0(
+      appDeployments$name,
+      " (", accountId(appDeployments$account, appDeployments$server), ")"
+    )
+    not_interactive <- c(
+      "Please use {.arg appName}, {.arg server} or {.arg account} to disambiguate.",
+      i = "Known application names: {.str {unique(appDeployments$name)}}.",
+      i = "Known servers: {.str {unique(appDeployments$server)}}.",
+      i = "Known account names: {.str {unique(appDeployments$account)}}."
+    )
+    idx <- cli_menu(
+      "This directory has been previously deployed in multiple places.",
+      "Which deployment do you want to use?",
+      choices = choices,
+      not_interactive = not_interactive,
+      error_call = error_call
+    )
+    createDeploymentTarget(
+      appDeployments$name[[idx]],
```

Comment:
Does it make sense to use a temporary and `xx[[idx]]` once rather than seven distinct indexing operations? This is effectively the same code as on line 52; is there an obvious way to avoid that repetition?

Reply:
Maybe the way forward is to add a function that updates an existing deployment record. Let me try that and see how it feels.